Deep fried inedible items are moved to stomach when eaten

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

AnturK said it was okay to make another PR for https://github.com/tgstation/tgstation/pull/64747.

### Quick rundown/basic gestalt:
- Deep fried inedible items are moved to stomach when eaten, instead of being deleted
- Having inedible items in stomach increases disgust and damages stomach at random intervals based on their total size
- Vomiting empties stomach
- Stomachs act like (worthless) bags now
- Default stomachs hold a combined weight class of 12 (only for consumed items), each tier of cybernetic stomach carries more
- Toy singulo suicide has been updated to use the stomach's contents instead of cavity implants

### Some considerations:

**Q: Can this be used to create infinite food?**
A: No, vomiting causes more nutrition loss than you can offset with a few deep fried inedible objects.

**Q: Can this be used to carry the armoury?**
A: Vomiting stuns you, so the other person would get access to the vomited guns first. You could use a monkey with the tier 3 stomach to carry 6 bulky guns, but they'd vomit them up in less than a minute.

**Q: Given organs are small items (fit in boxes), does this make stomachs better storage items than smuggler's satchels?**
A: No, stomachs only hold two tiny items. Eating ignores this limit.

**Q: Does this mean I could forcefeed a monkey deep fried shotguns, remove its stomach, and carry it around in my box?**
A: No, removing stomachs drop their contents.

**Q: Does this affect stomachs of species that don't eat?**
A: Species that don't eat can still eat, so I've only removed it from Ethereal stomachs. They're called biological batteries and given they don't eat I find it hard to justify something called that holding eaten things. The rest are essentially still stomachs.

**Q: Aren't deep fried objects unusable?**
A: The deep fry holder, which is an object that mimics the deep fried object, contains the deep fried object. Before this PR, the deep fried object was deleted when the deep fry holder was eaten. Now, the object is moved to the stomach.

**Q: What happens when mobs are consumed?**
A: This works similarly to being held as a mob. You see and hear what the mob holding you sees. You can be heard. Eventually you're vomited back up. The alternative is preventing objects with clients in them from being deep fried, but that's boring.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

It's cool and adds more depth  
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Deep fried inedible items are moved to stomach when eaten, instead of being deleted
add: Having inedibles in stomach increases disgust and damages stomach at random intervals based on their total volume
add: Having too many items in stomach triggers vomiting
add: Vomiting empties stomach
add: Stomachs act like (worthless) bags
expansion: Toy singulo suicide uses stomach contents instead of cavity implants
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Eating deep fried inedibles moves them to stomach

- Having inedibles in stomach every so often increases disgust based on the amount
- Vomiting empties stomach contents
- Can squeeze out contents of stomachs in hand

cacogen committed:
length()

Co-authored-by: Zonespace <41448081+Zonespace27@users.noreply.github.com>

cacogen committed:
Adds some boring and unnecessary checks for volume

cacogen committed:
Makes spinning toy suicide work with this

- Adds stomach damage at random intervals based on volume held in stomach
- Adds autovomit when stomach is too full

cacogen committed:
Sins of a solar empire

cacogen committed:
how did you get your early return game to come up

cacogen committed:
Improves timing of singulo toy suicide

- Adds feedback for emptying empty stomach
- Adds examine message explaining how to empty stomachs

cacogen committed:
Different volumes for different stomachs

This feels like making my own shitty imitation storage object on the side of real storage

cacogen committed:
Makes it use the storage component

Y0SH1M4S73R commented:
You know what would be a hilarious follow-up to this PR? A pica mutation that prevents the disgust gain and/or the stomach damage.

cacogen committed:
Remembers to QDEL_LIST deepfry holder contents if we can't use it or we're done with it

- Gets rid of comment I initially preserved to be respectful to its author but fails to be funny

nianjiilical commented:
make sure stomach contents are checked for steal objectives so we can just eat the hand teleporter

Ryll-Ryll commented:
Wow, this is a far, far more fleshed out PR than it was I first looked at the old one. Very interested!

ErdinyoBarboza commented:
full circle back to vorecode

cacogen committed:
Length

cacogen commented on code/modules/surgery/organs/stomach/_stomach.dm:
```diff
@@ -39,6 +43,31 @@
 	else
 		reagents.flags |= REAGENT_HOLDER_ALIVE
 
+/obj/item/organ/stomach/ComponentInitialize()
```
So, what should I use instead?

MrMelbert replied:
You can just put it in `Initialize()`

TemporalOroboros commented:
Along the same lines as the singularity toy suicide you might want to take a look at the grenade suicide.

oklews commented:
aren't deepfried objects unusable? Are you like removing the deepfried status from it when it exits the stomach? Could you make eating sharp objects cause bleeding?

cacogen committed:
Puts stomach in mob to enable fun

- Simplifies the disgust accumulation calculation for predictability's sake (I balanced this around the consumption of a single intellicard and found that it vomits within 2 minutes of consumption with a normal stomach and about 4 minutes with a tier 3 one, not taking into account things a player can use to lower disgust).
- Makes sharp objects do extra stomach damage (Sharp Objects was a good show btw)

cacogen committed:
Allows you to cough blood

oklews commented:
> aren't deepfried objects unusable? Are you like removing the deepfried status from it when it exits the stomach? Could you make eating sharp objects cause bleeding?

Local man doesn't read the PR

oklews commented:
would you be open to making formerly undeepfriable things deepfriable?

cacogen commented:
I dunno, making things like the nuke disk consumable seems like it's for a different PR. There are still issues with this one that mean it might not be merged, namely the consequences of putting the stomach in the mob or how limited the feature becomes if I have to leave the organ in nullspace. You couldn't eat the disk anyway if I had to do that, because it would immediately move.

interestingusernam3 commented:
vore

oklews commented:
> I dunno, making things like the nuke disk consumable seems like it's for a different PR. There are still issues with this one that mean it might not be merged, namely the consequences of putting the stomach in the mob or how limited the feature becomes if I have to leave the organ in nullspace. You couldn't eat the disk anyway if I had to do that, because it would immediately move.

Things like the HOS laser or the Captain's laser aren't deep friable

cacogen commented:
I'm going to close this because I have too many PRs to focus on at the moment and I don't think this is going to be merged because of the potential for bugs from putting the stomach into the mob.
